server/src/main/java/io/crate/action/sql/Cursor.java

```diff
@@ -62,11 +64,13 @@ public Cursor(CircuitBreaker circuitBreaker,
                   boolean scroll,
                   Hold hold,
                   CompletableFuture<BatchIterator<Row>> queryIterator,
-                  List<Symbol> outputs) {
+                  List<Symbol> outputs,
+                  AnalyzedStatement query) {
```

Comment:
Looks like this change is no longer necessary?

The privilege check uses the query on `AnalyzedDeclare`, not on the `Cursor` ?

Reply:
Yes, you are right. Thank you!